backend/infrahub/graphql/mutations/main.py

```diff
@@ -189,6 +204,40 @@ async def mutate_create_to_graphql(cls, info: GraphQLResolveInfo, db: InfrahubDa
             result["object"] = await obj.to_graphql(db=db, fields=fields.get("object", {}))
         return cls(**result)
 
+    @classmethod
+    async def _call_mutate_update(
+        cls,
+        info: GraphQLResolveInfo,
+        data: InputObjectType,
+        branch: Branch,
+        db: InfrahubDatabase,
+        obj: Node,
+    ) -> tuple[Node, Self]:
+        """
+        Wrapper around mutate_update to potentially activate locking and call it within a database transaction.
+        """
+
+        lock_kinds = get_kinds_to_lock_on_object_mutation(cls._meta.schema)
```

Comment:
Would be good to have `get_kinds_to_lock_on_object_mutation` (or another function) return the lock names directly